src/loaders/compute_tools/tool_common.py

```diff
@@ -449,21 +450,26 @@ def create_metadata_file(
 # other ways. Meh
 def _prepare_tool(
         work_dir: Path,
-        batch_id: str,
-        node_id: str,
+        job_id: str,
         data_ids: List[str],
         source_data_dir: Path,
         source_file_ext: str,
         allow_missing_files: bool,
         data_id_from_filename: bool,
         suffix_ids: bool,
+        batch_tool: bool,
 ) -> Tuple[Path, Dict[str, Dict[str, Union[str, Path]]]]:
     # Prepares for tool execution by creating a batch directory and retrieving data
     # files with associated metadata.
 
-    # create working directory for each batch
-    batch_dir = work_dir / (f'{loader_common_names.COMPUTE_OUTPUT_PREFIX}{batch_id}_size_'
-                            + f'{len(data_ids)}_node_{node_id}')
+    # create a working directory for each job
+    # parser program will need `COMPUTE_OUTPUT_PREFIX` as dir name prefix to identify compute directories
+    # and `COMPUTE_OUTPUT_NO_BATCH` as dir name suffix to identify non-batch directories
+    batch_job_dir_identifier = '' if batch_tool else loader_common_names.COMPUTE_OUTPUT_NO_BATCH
+    batch_dir_name = (f'{loader_common_names.COMPUTE_OUTPUT_PREFIX}_{job_id}'
+                      f'_size_{len(data_ids)}{batch_job_dir_identifier}')
+    batch_dir = work_dir / batch_dir_name
```

Comment:
Does this mean the parser can no longer handle data from older runs?

Reply:
yes. But if we need to we can quickly update `COMPUTE_OUTPUT_PREFIX` and `COMPUTE_OUTPUT_NO_BATCH`. Then run the parser program. Or we can keep them as is. The dir name will be something like `batch_0_size_1000_no_batch` or `batch_0_size_1000` which is also acceptable.

Reply:
vs `job_0_size_1000_non_batch` or `job_0_size_1000`

Reply:
Would it be difficult to rename the directories of older runs?

Reply:
oh yea. not at all. We can rename old dirs.

Reply:
:+1: